### PR TITLE
Menu images fetcher

### PR DESCRIPTION
**Pull request by XiaoXuxxxx**
- add function getShopLinkById() this function will return shop's link that relate to food id

- and api that run it 

- improved/fix function about database manipulation

### CODE REVIEW

File: server/models/menu.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+const { where } = require('../database/knex.js');
```
Comment (riflowth) [inline server/models/menu.js]: Unnecessary destructuring

File: server/models/menu.js [hunk 2]
```diff
@@ -28,30 +29,39 @@ exports.findLastId = async (req, res) => {
...
+exports.updateFoodData = async (id, name, type, price , shop_id) => {
+    if (id == 0 || id == undefined) id = await this.findLastId();
+    await knex('foods') //นี่ก็ด้วยยย
         .where({ id: id })
         .update({
             name: name,
             type: type,
             price: price,
-            image_url: image_url,
+            image_url: `../../public/images/menus/shopid${shop_id}_menu${id}`,
```
Comment (riflowth) [inline server/models/menu.js]: Unnecessary prefix why not `number_number`

File: server/models/menu.js [hunk 3]
```diff
@@ -28,30 +29,39 @@ exports.findLastId = async (req, res) => {
...
             shop_id: shop_id
         });
+}
+//TODO
+exports.getShopLinkByImage = async (id) => {
+    let shop_id = await knex('foods').select('shop_id').where({ id: id })
+    shop_id = Object.values(JSON.parse(JSON.stringify(shop_id[0])))[0];
+    let shop_link = `http://localhost:8080/shop/${shop_id}`
```
Comment (riflowth) [inline server/models/menu.js]: What wrongs with this if the server is not up to localhost and 8080 port?

File: server/models/menu.js [hunk 4]
```diff
@@ -28,30 +29,39 @@ exports.findLastId = async (req, res) => {
...
+        image_url: `../../public/images/menus/shopid${shop_id}_menu${id}`,
         shop_id: shop_id
     }).into('foods');
 }
 
 exports.deleteFoodData = async (id) => {
-    if (id == 0) id = await this.findLastId();
-    knex('foods').where({ id: id }).del();
+    if (id == 0 || id == undefined) id = await this.findLastId();
+    await knex('foods').where({ id: id }).del(); //มันต้องมีawaitอะ ไม่งั้นก็ไม่ลบ ไม่รู้ทำไม
```
Comment (riflowth) [inline server/models/menu.js]: Because this is a promise

File: server/controllers/api.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ exports.addFoodData = (req, res) => {
     let price = req.body.price;
     let shop_id = req.body.shop_id;
     Menu.insertFoodData(id, name, type, price, shop_id);
-  
+    console.log("hi");
```
Comment (riflowth) [inline server/controllers/api.js]: Hello! ?

File: server/models/menu.js [hunk 6]
```diff
@@ -28,30 +29,39 @@ exports.findLastId = async (req, res) => {
...
             name: name,
             type: type,
             price: price,
-            image_url: image_url,
+            image_url: `../../public/images/menus/shopid${shop_id}_menu${id}`,
             shop_id: shop_id
         });
+}
+//TODO
+exports.getShopLinkByImage = async (id) => {
```
Comment (riflowth) [inline server/models/menu.js]: Bad function naming **id** is a parameter but this function named `ByImage`

**Comment by XiaoXuxxxx** [comment]
Thank you

**Comment by riflowth** [comment]
@XiaoXuxxxx Thanks for improving readyyy to mergeeeee :rocket:

**Pull request merged**